result.go

```diff
@@ -546,17 +546,46 @@ func (q Query) ToMapByT(result interface{}, keySelectorFn interface{}, valueSele
 	q.ToMapBy(result, keySelectorFunc, valueSelectorFunc)
 }
 
-// ToSlice iterates over a collection and populates the result slice with elements.
-// Collection elements must be assignable to the slice's element type.
-// ToSlice doesn't empty the result slice before populating it.
-func (q Query) ToSlice(result interface{}) {
-	res := reflect.ValueOf(result)
+// ToSlice iterates over a collection and saves the results in the slice pointed
+// by v. It starts saving elements starting from index 0, therefore it will
+// overwrite the slice.
+//
+// If the slice pointed by v has sufficient capacity, v will be pointed to a
+// resliced slice. If it does not, a new underlying array will be allocated and
+// v will point to it.
+func (q Query) ToSlice(v interface{}) {
+	res := reflect.ValueOf(v)
 	slice := reflect.Indirect(res)
 	next := q.Iterate()
-
+	index := 0
 	for item, ok := next(); ok; item, ok = next() {
-		slice = reflect.Append(slice, reflect.ValueOf(item))
+		if index >= slice.Len() {
+			slice = grow(slice, 1)
```

Comment:
Actually, we can do this in another follow-up PR. Let me know if you want to keep working on the current PR to implement the proposed algorithm, or if you want this to just get merged and you can open a PR later.